UofT-Frosh-Orientation/orientation-website
# 353 edit profile page for non registered users v2

## **Contribution:**


### **Issue:**

Closes #353 

### **Accomplishments:**

- Edited existing code for `ProfileEdit.js` page to correctly work only if a user is unregistered, otherwise redirect to profile page, and only show the 'edit' icon if they are unregistered
- Added function updateInfo in `UserController.js` to handle updating information for a user given some changed userModel properties, in this case firstName, lastName and preferredName
- Added route in `userRoutes.js` with a put request for updating user information based on submission of form. 
- Used axios instance within `ProfileEdit.js` page to change information on submission, since it's not using dispatch currently, this change takes effect after the user logs in again.

### **Visuals:** 

![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/79293445/273b97c9-6726-49a8-a43c-0ced3957f9a0)
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/79293445/7443736f-1fb1-4ad7-95c9-757f809d2f4c)
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/79293445/8d9f1cf2-c225-412a-9db1-ba478394a2d6)



## **Details:**

### **Expected Behavior:**

- Should see a yellow icon for editing (pencil) on the profile page
- When clicked, it will redirect to the profile edit page, where new values can be entered for first, last, and preferred names.
- When the form is submitted, it should send you back to the profile page.
- The next time you log in, the names will be updated.

### **Testing Steps:**

Steps required to get expected behavior.

- Load website with readyForFrosh = true
- Create an account (Frosh/Leedur, both are fine)
- Head to the profile page and make sure that the yellow icon pops up. Click it to edit names
- Try changing first, last and preferred names (maybe leave some blank and see if they give any weird behavior) and submit the form
- Sign out and log back in to see updated names, if working correctly.

### **Complications:**

- At times, getting Axios:Network Error. Though I have found no real solution when it starts to happen, usually I troubleshoot by restarting my computer, deleting node_modules and mongo_volumes, and also the docker containers/images. After reinstalling some/all parts it usually ends up working.
- `nodemon app crashed - waiting for file changes` was another error I got in the API container. It disappeared early on though. 
- Issue committing code in server folder, with it saying `husky - pre-commit hook exited with code 127 (error)`. Fixed with Uzma's help and using `git config--unset core.hooksPath`

### **Resources:**

Any additional resources used outside this in issue.

- Committing issues: [(https://github.com/https://github.com/typicode/husky/issues/1038)] (Uzma's Link)
- [Express JS tutorial](https://www.youtube.com/watch?v=SccSCuHhOw0&ab_channel=WebDevSimplified)
- [Middleware Explainer Video](https://www.youtube.com/watch?v=lY6icfhap2o)
